service/routers/decks.js

```diff
@@ -18,69 +18,144 @@ const getDecks = async (req, res) => {
     console.log(`${getDecks.name}: ${err}`)
     res.sendStatus(500)
   }
-}
+} 
 
 const createDeck = async (req, res) => {
-  const userId = ''
   const newDeck = req.body
-  try {
-    const user = await User.findById(userId)
-    user.decks.push({
-      name: newDeck.name,
-      cards: []
-    })
-    await user.save()
-    res.sendStatus(204)
-  } catch (err) {
-    console.log(`${createDeck.name}: ${err}`)
+  const { userId } = req.user
+  const requestor = await User.findById(userId)
+  if (requestor.role === 'admin' || requestor.role === 'superuser') {
+    try {
+      const user = await User.findById(userId)
+      user.decks.push({
+        name: newDeck.name,
+        cards: []
+      })
+      await user.save()
+      res.sendStatus(204)
+    } catch (err) {
+      console.log(`${createDeck.name}: ${err}`)
+      res.sendStatus(500)
+    }
+  } else if (requestor.role === 'user') {
+    try {
+      const user = await User.findById(req.params.id)
+      if (user.role = 'user') {
```

Comment:
You should treat any suggestion of information from the client as suspicious. Do you really need to use this? What should you be checking instead?